File: EcoBookAiBackend/src/main/java/com/ecobook/dto/MaterialDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class MaterialDTO {
    private String id;
    private String titulo;
    private String autor;
    private String editora;
    private String descricao;
    private String disciplina;
    private String nivelEnsino;
    private Integer ano;
    private String sistemaEnsino;
    private String estadoConservacao;
    private String status;
    private String imagemUrl;
    private String uploadId;
    private MaterialDonorDTO doador;
    private String cidade;
    private String bairro;
    private Integer dataPublicacao;
    private String statusIa;
    private Double confiancaIa;
    private LocalDateTime criadoEm;
    private LocalDateTime atualizadoEm;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/MaterialDTO.kt
package com.ecobook.dto

data class MaterialDTO(
    val id: String,
    val titulo: String,
    val descricao: String,
    val disciplina: String,
    val nivelEnsino: String,
    val ano: Int?,
    val sistemaEnsino: String,
    val estadoConservacao: String,
    val status: String,
    val imagemUrl: String?,
    val uploadId: String?,
    val doador: MaterialDonorDTO,
    val cidade: String,
    val bairro: String,
    val dataPublicacao: Int?,
    val statusIa: String?,
    val confiancaIa: Double?,
    val criadoEm: String?,
    val atualizadoEm: String?
)
